### ISSUE TITLE
Option for "Do Not Overwrite Target Field Value with Null Value"

### ISSUE THREAD

**Issue by sd5000** [afawcett/declarative-lookup-rollup-summaries#521]
It would be great to have a configuration option to tell the rollup process not to delete an existing value in the target field if there is nothing to roll up. Essentially saying "if you have an updated value then update the target field, but otherwise leave it alone."

A sample use case is that we have a field on Contact called "Date of Last Soft Credit". DLRS is set up to roll up the max date of child soft credit records to that field. That works well.

However some Contact records already have a legacy date in that field that we want to keep, even though they _don't_ have corresponding soft credit objects. Currently DLRS deletes that legacy date value when it batch calculates the rollups, since there are no child soft credit records. It would be great to be able to tell it not to do that.

Obviously this wouldn't be the ideal default behavior, but to be able to turn it on would be super useful. I've seen similar settings in tools like formassembly (screenshot below), dataloader.io, etc.

<img width="464" alt="donotoverwrite" src="https://user-images.githubusercontent.com/2791397/27240714-c1b1d726-528a-11e7-8126-bc238afe893f.png">

**Comment by afawcett** [comment]
Yes @sd5000 thanks for this suggestion, its been covered before in another enhancement request, so this maybe a dupe, for now though i'll leave it open. Thanks again!

**Comment by asinclair56** [comment]
@afawcett where was this feature implemented, because I'm getting populated values nulled by the tool in version 2.11

**Comment by mazz44** [comment]
This would be a helpful feature. I recently had to roll up to a newly created field since this option was not available.

**Comment by tcharles12** [comment]
> Yes @sd5000 thanks for this suggestion, its been covered before in another enhancement request, so this maybe a dupe, for now though i'll leave it open. Thanks again!

Any update on this item? Thanks!

**Comment by tcharles12** [comment]
> Yes @sd5000 thanks for this suggestion, its been covered before in another enhancement request, so this maybe a dupe, for now though i'll leave it open. Thanks again!

Any update on this item? Thanks!

**Comment by jenxyz** [comment]
Hello, this enhancement request 'Do Not Overwrite Target Field Value with Null Value' is affecting me - wondering if there is any update?

**Comment by aheber** [comment]
Similar to #161, different need but similar engine changes. Mentioning to link them together.
